### DISCUSSION TITLE
Interceptor to ignore server request

### DISCUSSION THREAD

**Discussion by mars3142** [lejard-h/chopper#347] [Q&A]
I want to build a preview app, where I can inject static json responses. Now my question. What can I do to stop the request to the server and return this json as a response? I don't want the app to do a server call, because the result will be thrown away.

A ResponseInterceptor can override my server result, but this isn't needed. 

Any ideas for that?

**Comment by techouse** [comment]
You should mock your `http` client and return specific static JSON for a specific URL.

Check this https://hadrien-lejard.gitbook.io/chopper/faq#mock-chopperclient-for-testing and this https://docs.flutter.dev/cookbook/testing/unit/mocking#3-create-a-test-file-with-a-mock-httpclient
